gtars/src/overlap/traits.rs

```diff
@@ -4,11 +4,18 @@ pub use crate::common::models::Interval;
 
 pub trait Overlapper<I, T>: Send + Sync
 where
-    I: PrimInt + Unsigned + Ord + Clone + Send + Sync,
-    T: Eq   + Clone    + Send + Sync,
+    I: PrimInt + Unsigned + Clone + Send + Sync,
+    T: Eq + Clone + Send + Sync,
 {
-    fn build(intervals: Vec<Interval<I, T>>) -> Self  where Self: Sized;
-    // TODO: how to make this an iterator instead? itearting
-    // over found intervals, i believe, is more performant.
+    fn build(intervals: Vec<Interval<I, T>>) -> Self
+    where
+        Self: Sized;
+
     fn find(&self, start: I, end: I) -> Vec<Interval<I, T>>;
-}
\ No newline at end of file
+
+    fn find_iter<'a>(
+        &'a self,
+        start: I,
+        end: I,
+    ) -> Box<dyn Iterator<Item = &'a Interval<I, T>> + 'a>;
```

Comment:
Are you able to explain why `Box` is needed here? A box'd dyn makes sense since the compiler cant possibly know which is being called at compile time since its decided at runtime (i.e. dynamic dispatch), but the Iterator is a standard interface right?